spectrum-offchain-cardano/src/data/cfmm_pool.rs

```diff
@@ -836,15 +1063,318 @@ impl ApplyOrder<ClassicalOnChainRedeem> for ConstFnPool {
+
+        self.royalty_y = self
+            .royalty_y
+            .checked_sub(&order.withdraw_royalty_y)
+            .ok_or(ApplyOrderError::incompatible(royalty_withdraw.clone()))?;
+
+        self.reserves_x = self
+            .reserves_x
+            .checked_sub(&order.withdraw_royalty_x)
+            .ok_or(ApplyOrderError::incompatible(royalty_withdraw.clone()))?;
+
+        self.reserves_y = self
+            .reserves_y
+            .checked_sub(&order.withdraw_royalty_y)
+            .ok_or(ApplyOrderError::incompatible(royalty_withdraw.clone()))?;
+
+        self.nonce += 1;
+
+        let royalty_output = RoyaltyWithdrawOutput {
+            token_x_asset: self.asset_x,
+            token_x_amount: order.withdraw_royalty_x,
+            token_y_asset: self.asset_y,
+            token_y_amount: order.withdraw_royalty_y,
+            ada_residue: order.init_ada_value - order.fee,
+            redeemer_pkh: order.royalty_pub_key_hash,
+        };
+
+        let blueprint = OperationResultBlueprint {
+            outputs: OperationResultOutputs::SingleOutput(royalty_output),
+            witness_script: Some((
+                pool_order_validator.clone().erased(),
+                Box::new(move |(pool_id, order_idx)| {
+                    ConstrPlutusData::new(
+                        0,
+                        vec![
+                            PlutusData::Integer(BigInteger::from(pool_id)),
+                            PlutusData::Integer(BigInteger::from(order_idx)),
+                        ],
+                    )
+                    .into_pd()
+                }),
+            )),
+            order_script_validator: order_validator,
+            strict_fee: Some(royalty_withdraw_context.execution_fee),
+        };
+
+        Ok((self, blueprint))
+    }
+}
+
+impl<Ctx> ApplyOrder<OnChainDAOActionRequest, Ctx> for ConstFnPool
+where
+    Ctx: Has<DeployedValidator<{ RoyaltyPoolDAOV1 as u8 }>>
+        + Has<DeployedValidator<{ RoyaltyPoolDAOV1Request as u8 }>>
+        + Has<DeployedValidator<{ RoyaltyPoolV1 as u8 }>>
+        + Has<DAOContext>
+        + Has<NetworkId>,
+{
+    type Result = DaoActionResult;
+
+    fn apply_order(
+        mut self,
+        dao_request: OnChainDAOActionRequest,
+        ctx: Ctx,
+    ) -> Result<(Self, OperationResultBlueprint<DaoActionResult>), ApplyOrderError<OnChainDAOActionRequest>>
+    {
+        let validator = dao_request.get_validator(&ctx);
+        let dao_validator: DeployedValidator<{ RoyaltyPoolDAOV1 as u8 }> = ctx.get();
+        let pool_validator: DeployedValidator<{ RoyaltyPoolV1 as u8 }> = ctx.get();
+        let dao_ctx: DAOContext = ctx.get();
+
+        let data_to_sign: DaoRequestDataToSign = (
+            dao_request.clone().order,
+            pool_validator.hash,
+            self.nonce,
+            self.treasury_x.untag(),
+            self.treasury_y.untag(),
+        )
+            .into();
```

Comment:
Why not use a constructor simply? Tuple5 is a bit too complicated

Reply:
done